### PR TITLE
Show block reason on reviewer tools page

### LINKED ISSUE #16444: Show block reason on reviewer tools page
### Describe the problem and steps to reproduce it:

When an add-on is blocked, we show an entry in the version history on the reviewer tools page. That entry should also include the reason for blocking.

### What happened?

blocka

### What did you expect to happen?

Reason should be included, like so:

blockb

### Anything else we should know?
(Please include a link to the page, screenshots and any relevant files.)

### PR DESCRIPTION
**Pull request by wagnerand**
Fix #16444.

### CODE REVIEW

File: src/olympia/blocklist/utils.py [hunk 1]
```diff
@@ -69,7 +69,7 @@ def block_activity_log_delete(obj, *, submission_obj=None, delete_user=None):
         'url': obj.url,
         'reason': obj.reason,
         'include_in_legacy': obj.in_legacy_blocklist,
-        'comments': f'Versions {obj.min_version} - {obj.max_version} blocked.',
+        'comments': f'Versions {obj.min_version} - {obj.max_version} unblocked.',
```
Comment (wagnerand) [inline src/olympia/blocklist/utils.py:72]: Drive-by fix that had confused me in the past.

File: src/olympia/reviewers/templates/reviewers/includes/history.html [hunk 2]
```diff
@@ -9,6 +9,9 @@
     {% if record.details %}
       {# <pre> rather than <div> so that white space is preserved on copy #}
       <pre class="light history-comment">{{ record.details.comments }}</pre>
+        {% if record.details.reason %}
```
Comment (eviljeff) [inline src/olympia/reviewers/templates/reviewers/includes/history.html:12]: nit: unnecessary indentation

File: src/olympia/reviewers/templates/reviewers/includes/history.html [hunk 3]
```diff
@@ -9,6 +9,9 @@
...
+          <pre class="light history-comment">Reason: {{ record.details.reason }}</pre>
```
Comment (eviljeff) [inline src/olympia/reviewers/templates/reviewers/includes/history.html:13]: Can you attach a screenshot to the PR - I'm interested how to this actually looks (does it just get concatenated together with a space when rendered; line breaks are okay;etc).

**Review by eviljeff — CHANGES_REQUESTED** [review]
Needs a test 😛

File: src/olympia/reviewers/templates/reviewers/includes/history.html (hunk 3 above)
Comment (wagnerand) [reply src/olympia/reviewers/templates/reviewers/includes/history.html:13]: There is a screenshot in the issue. Is that sufficient?

Comment (eviljeff) [reply src/olympia/reviewers/templates/reviewers/includes/history.html:13]: Hah, you filed the issue _after_ you'd completed the patch.  Sure - I didn't look there.

File: src/olympia/reviewers/tests/test_views.py [hunk 4]
```diff
@@ -6042,11 +6043,16 @@ def test_blocked_versions(self):
         assert span.text() == 'Blocked Blocked'
         assert span.length == 2  # a new version is blocked too
 
-        block.update(max_version='98')
+        block_reason = 'Very bad addon!'
+        block.update(max_version='98', reason=block_reason)
+        block_activity_log_save(obj=block, change=False)
```
Comment (eviljeff) [inline src/olympia/reviewers/tests/test_views.py:6048]: I meant more calling `log_create` directly but this works too

**Review by eviljeff — APPROVED** [review]
